### PR TITLE
RR-160 - Retrieve Action Plan and place in overview screen "view data"

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR completes the work for retrieving a prisoner's Action Plan and putting it in the overview screen's "view data", but stops short of displaying it on the overview screen, because we don't yet have an agreed/updated design (sub-task https://dsdmoj.atlassian.net/browse/RR-162).

### CODE REVIEW

File: integration_tests/mockData/actionPlanByPrisonNumberData.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                status: 'NOT_STARTED',
+                sequenceNumber: 2,
+              },
+            ],
+            createdBy: 'auser_gen',
+            createdAt: '2023-07-20T09:29:15.386Z',
+            updatedBy: 'auser_gen',
+            updatedAt: '2023-07-20T09:29:15.386Z',
+            reviewDate: null,
+            notes: "Pay close attention to Paris' behaviour.",
```
Comment (chrimesdev) [inline integration_tests/mockData/actionPlanByPrisonNumberData.ts:38]: This seems to be *note* singular in the template - need to double check

Comment (mattwills-moj-digital) [reply integration_tests/mockData/actionPlanByPrisonNumberData.ts:38]: That's right, it's `notes` in the API, but `note` in the UI project.

File: integration_tests/mockData/actionPlanByPrisonNumberData.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+              {
+                stepReference: '177e45eb-c8fe-438b-aa81-1bf9157efa05',
+                title: 'Book French course',
+                targetDateRange: 'THREE_TO_SIX_MONTHS',
+                status: 'NOT_STARTED',
+                sequenceNumber: 1,
+              },
+              {
+                stepReference: '32992dd1-7dc6-4480-b2fc-61bc36a6a775',
+                title: 'Complete French aid course',
```
Comment (nathanrussell-moj-digital) [inline integration_tests/mockData/actionPlanByPrisonNumberData.ts:69]: Very very minor (I know its only test data! 🤣 ) - but I guess this is copy and paste from the first aid course above, unless "french aid" is something entirely different! 😁

File: integration_tests/mockData/actionPlanByPrisonNumberData.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
```
Comment (nathanrussell-moj-digital) [inline integration_tests/mockData/actionPlanByPrisonNumberData.ts:138]: More French aid! 🤣

File: server/routes/overview/overviewController.ts [hunk 4]
```diff
@@ -3,17 +3,22 @@ import EducationAndTrainingView from './educationAndTrainingView'
...
+    private readonly curiousService: CuriousService,
+    private readonly educationAndWorkPlanService: EducationAndWorkPlanService,
+  ) {}
 
   getOverviewView: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber } = req.params
     req.session.createGoalForm = undefined
 
     const { prisonerSummary } = req.session
+    const actionPlan = await this.educationAndWorkPlanService.getActionPlan(prisonNumber, req.user.token)
```
Comment (nathanrussell-moj-digital) [inline server/routes/overview/overviewController.ts:19]: Excellent - we are using the user token for this API call 👍 Thats the correct one to use in this case 👍

File: integration_tests/mockData/actionPlanByPrisonNumberData.ts (hunk 2 above)
Comment (mattwills-moj-digital) [reply integration_tests/mockData/actionPlanByPrisonNumberData.ts:69]: How dare you! The prisoner wishes to learn how to distribute aid as part of their bid to join the French Foreign Legion 😉

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍 
Great work, both of you @mattwills-moj-digital @chrimesdev  👍

File: integration_tests/mockData/actionPlanByPrisonNumberData.ts (hunk 2 above)
Comment (mattwills-moj-digital) [reply integration_tests/mockData/actionPlanByPrisonNumberData.ts:69]: I'll fix it in my next PR. If I make the change now then I'll need a re-approval! 😩

Comment (mattwills-moj-digital) [reply integration_tests/mockData/actionPlanByPrisonNumberData.ts:69]: Oh, Adam's already merged it anyway! 😂

**Pull request merged**